src/CIFAR10/CIFAR10.jl

```diff
@@ -59,13 +45,13 @@ module CIFAR10
 
     This function will display an interactive dialog unless
     either the keyword parameter `i_accept_the_terms_of_use` or
-    the environment variable `DATADEPS_ALWAY_ACCEPT` is set to
+    the environment variable `DATADEPS_ALWAYS_ACCEPT` is set to
```

Comment:
good catch!

Reply:
> It makes it harder to review and I don't think there is a reason to do that change. Could you please change that back?

Makes sense. I've restored the file structure.

However, in the long run I think these changes make sense. `Reader` folder have only one `Reader.jl` file; `reader*.jl` files in `MNIST` are very small, also reading images/labels is coupled with reading headers, so I think there's a good reason to keep these functions in one file. Personally, I would also simplify the modules structure, because it makes sharing the code (e.g. between CIFAR10 and CIFAR100) harder, also this boundaries are a bit artificial (e.g. CIFAR10/interface.jl contains the logic to decide from which file to read from).
Of course it makes sense to define and export the modules for the individual datasets, but they may be very small and just contain the `MNIST.traindata/testdata()` public wrappers, while the actual private functions would reside in one single MLDatasets module.

Reply:
I agree there is an argument to be had to simplify the structure, but this should be separate from this PR. 

You do a lot of objectively important updates in this PR that I am very excited about. Let us not hold those back with subjective preferences.